manifests/scheduled_repo.pp

```diff
@@ -0,0 +1,38 @@
+# @summary A wrapper that creates a pulp_Xrepo resource with and an associated pulp_schedule resource
+#
+# @example Basic usage with default `rpm` `repo_type` and 'random' daily schedule
+#   pulp::scheduled_repo { 'centos-7':
+#     repo_config => {
+#       'display_name' => 'CentOS 7 Base Repo',
+#       'feed'         => 'https://www.mirrorservice.org/sites/mirror.centos.org/7/os/x86_64',
+#     },
+#   }
+#
+# @param repo_type
+#   Determines whether a `pulp_rpmrepo`, `pulp_puppetrepo` or `pulp_isorepo` resource is created.
+# @param repo_config
+#   Defines a hash of parameters to pass to the `pulp_Xrepo` resource.
+# @param schedule
+#   An iso8601 schedule string or the special value 'daily'. Defaults to 'daily' which creates a `pulp_schedule` with a pseudo-random daily `schedule_time`.
+define pulp::scheduled_repo
+(
+  Enum['rpm','puppet','iso'] $repo_type = 'rpm',
+  Hash $repo_config = {},
+  Variant[Enum['daily'],String[1]] $schedule = 'daily',
```

Comment:
I appreciate `String[1]` already encompasses `'daily'`.  Hopefully this documents the intent though and one day someone might replace `String[1]` with an actual regex pattern to use for an iso8601 interval.

I suppose I could create a type alias (even if the implementation sucks).